crates/orchestrator/src/data_storage/aws_s3/mod.rs

```diff
@@ -88,7 +102,12 @@ impl DataStorage for AWSS3 {
 
     async fn create_bucket(&self, bucket_name: &str) -> Result<()> {
         if self.bucket_location_constraint.as_str() == "us-east-1" {
-            self.client.create_bucket().bucket(bucket_name).send().await?;
+            self.client
+                .create_bucket()
+                .bucket(bucket_name)
+                .send()
+                .await
+                .context(format!("Failed to create bucket: {} in us-east-1", bucket_name))?;
```

Comment:
Can we try to change this 
```
 if self.bucket_location_constraint.as_str() == "us-east-1" {
```
and compare the enums provided by AWS.

Accordingly change the `context` to not hardcode `us-east-1`

Note:  I tried doing this in a recent PR but the code was failing after
maybe you can try and this time it might work ?
WDYT @Mohiiit ?